requesthandler_test.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestHealthHandler_StatusOK_LockFilePresent(t *testing.T) {
+	rr := httptest.NewRecorder()
+
+	present := lockFilePresent()
+
+	if present == false {
+		if err := lock(); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	req, err := http.NewRequest(http.MethodGet, "/_/health", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	handler := makeHealthHandler()
+	handler(rr, req)
+
+	required := http.StatusOK
+	if status := rr.Code; status != required {
+		t.Errorf("handler returned wrong status code - want: %v, got: %v", required, status)
+	}
+
+}
+
+func TestHealthHandler_StatusInternalServerError_LockFileNotPresent(t *testing.T) {
+	rr := httptest.NewRecorder()
+
+	if lockFilePresent() == true {
+		if err := removeLockFile(); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	req, err := http.NewRequest(http.MethodGet, "/_/health", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	handler := makeHealthHandler()
+	handler(rr, req)
+
+	required := http.StatusInternalServerError
+	if status := rr.Code; status != required {
+		t.Errorf("handler returned wrong status code - want: %v, got: %v", required, status)
+	}
+}
+
+func TestHealthHandler_StatusMethodNotAllowed_ForWriteableVerbs(t *testing.T) {
+	rr := httptest.NewRecorder()
+
+	verbs := []string{http.MethodPost, http.MethodPut, http.MethodDelete}
+
+	for _, verb := range verbs {
+		req, err := http.NewRequest(verb, "/_/health", nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		handler := makeHealthHandler()
+		handler(rr, req)
+
+		required := http.StatusMethodNotAllowed
+		if status := rr.Code; status != required {
+			t.Errorf("handler returned wrong status code -  want: %v, got: %v", required, status)
+		}
+	}
+}
+
+func removeLockFile() error {
+	path := filepath.Join(os.TempDir(), ".lock")
+	log.Printf("Removing lock-file : %s\n", path)
```

Comment:
We shouldn't encourage logging messages in tests.

Reply:
There are methods in the `testing` package for this. `t.Log` and `t.Logf` which only get printed when a test fails or the `-test.v` flag is set. See [here](https://golang.org/pkg/testing/#T.Log).

Reply:
I have removed the log statement.